lib/src/models/data/home.dart

```diff
@@ -30,4 +33,41 @@ class HomeModel extends Model {
 		notifyListeners();
 		_messageTimer = Timer(const Duration(seconds: 5), () { message = null; notifyListeners(); });
 	} 
+
+  /// Starts a new timer
+  void startTimer({required MissionTimer timer}){
+    this.timer = timer;
+    runTimer();
+  }
+
+  /// Runs the timer
+  /// Updates the UI every second
+  void runTimer(){
+    if(timer!.paused || timer!.timeLeft <= Duration.zero){
+      notifyListeners();
+      return;
+    }
+    Timer(const Duration(seconds: 1), () {
+      notifyListeners();
+      runTimer();
+    });
+  }
+
+  /// Pauses the timer
+  void pauseTimer(){
+    timer!.remainingTime = timer!.timeLeft;
+    timer!.paused = true;
+  }
+
+  /// Resumes timer
+  void resumeTimer(){
+    timer!.paused = false;
+    startTimer(timer: MissionTimer(name: timer!.name, duration: timer!.remainingTime!));
+  }
+
+  /// Removes the timer from displaying on homepage
+  void stopTimer(){
+    timer = MissionTimer(name: "", duration: Duration.zero);
```

Comment:
```suggestion
    timer = null;
```
since you already handle that case in the UI?

Reply:
I remember trying to do this and I was getting null check errors 
solved now tho i think